<!-- build time:Sat Apr 06 2019 15:18:12 GMT+0800 (GMT+08:00) --><!DOCTYPE html><html class="theme-next gemini use-motion" lang="zh-CN"><head><meta name="generator" content="Hexo 3.8.0"><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=2"><meta name="theme-color" content="#1e90ff"><meta name="keywords" content="Android,iOS,Vue,Java,Swift,H5,HTML,JS,CSS,Perl,微信小程序,开发,博客,潘高"><script>!function(e,t,o,c,a,i,n){e.DaoVoiceObject=a,e[a]=e[a]||function(){(e[a].q=e[a].q||[]).push(arguments)},e[a].l=1*new Date,i=t.createElement(o),n=t.getElementsByTagName(o)[0],i.async=1,i.src=c,i.charset="utf-8",n.parentNode.insertBefore(i,n)}(window,document,"script",("https:"==document.location.protocol?"https:":"http:")+"//widget.daovoice.io/widget/0f81ff2f.js","daovoice"),daovoice("init",{app_id:"c5a5d9c7"}),daovoice("update")</script><link rel="stylesheet" href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2"><link rel="stylesheet" href="/css/main.css?v=7.0.0"><link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png?v=7.0.0"><link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png?v=7.0.0"><link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png?v=7.0.0"><link rel="mask-icon" href="/images/logo.svg?v=7.0.0" color="#1e90ff"><script id="hexo.configurations">var NexT=window.NexT||{},CONFIG={root:"/",scheme:"Gemini",version:"7.0.0",sidebar:{position:"right",display:"post",offset:12,b2t:!0,scrollpercent:!0,onmobile:!0},fancybox:!1,fastclick:!1,lazyload:!1,tabs:!0,motion:{enable:!0,async:!1,transition:{post_block:"fadeIn",post_header:"slideDownIn",post_body:"slideDownIn",coll_header:"slideLeftIn",sidebar:"slideUpIn"}},algolia:{applicationID:"",apiKey:"",indexName:"",hits:{per_page:10},labels:{input_placeholder:"Search for Posts",hits_empty:"We didn't find any results for the search: ${query}",hits_stats:"${hits} results found in ${time} ms"}}}</script><script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script><script>(adsbygoogle=window.adsbygoogle||[]).push({google_ad_client:"ca-pub-5317369003185250",enable_page_level_ads:!0})</script><meta name="description" content="常用生物信息学格式fasta,fastq,sra,gff2,gtf,gff3,bed,sam,bam,vcf的介绍。"><meta name="keywords" content="生物信息分析,格式介绍,fasta,fastq,sra,gff2,gtf,gff3,bed,sam,bam,vcf"><meta property="og:type" content="article"><meta property="og:title" content="常用生物信息学格式介绍"><meta property="og:url" content="https://blog.pangao.vip/常用生物信息学格式介绍/index.html"><meta property="og:site_name" content="潘高的小站"><meta property="og:description" content="常用生物信息学格式fasta,fastq,sra,gff2,gtf,gff3,bed,sam,bam,vcf的介绍。"><meta property="og:locale" content="zh-CN"><meta property="og:image" content="https://ws4.sinaimg.cn/large/006tKfTcgy1g0umxr08r8j30sg0fggqo.jpg"><meta property="og:image" content="https://ws3.sinaimg.cn/large/006tKfTcgy1g0ung2b9yhj30g005hwfa.jpg"><meta property="og:image" content="https://ws2.sinaimg.cn/large/006tKfTcgy1g0ungd9nx4j30ec05wt8v.jpg"><meta property="og:image" content="https://ws1.sinaimg.cn/large/006tKfTcgy1g0unj39xr5j30sg08e0ue.jpg"><meta property="og:image" content="https://ws3.sinaimg.cn/large/006tKfTcgy1g0unkjrkwwj30s30ajgny.jpg"><meta property="og:image" content="https://ws4.sinaimg.cn/large/006tKfTcgy1g0unlikupmj30nh0ea760.jpg"><meta property="og:updated_time" content="2019-03-07T15:41:49.428Z"><meta name="twitter:card" content="summary"><meta name="twitter:title" content="常用生物信息学格式介绍"><meta name="twitter:description" content="常用生物信息学格式fasta,fastq,sra,gff2,gtf,gff3,bed,sam,bam,vcf的介绍。"><meta name="twitter:image" content="https://ws4.sinaimg.cn/large/006tKfTcgy1g0umxr08r8j30sg0fggqo.jpg"><link rel="canonical" href="https://blog.pangao.vip/常用生物信息学格式介绍/"><script id="page.configurations">CONFIG.page={sidebar:""}</script><title>常用生物信息学格式介绍 | 潘高的小站</title><script>var _hmt=_hmt||[];!function(){var e=document.createElement("script");e.src="https://hm.baidu.com/hm.js?96a6d60d17907447305305696865fe41";var t=document.getElementsByTagName("script")[0];t.parentNode.insertBefore(e,t)}()</script><noscript><style>.sidebar-inner,.use-motion .brand,.use-motion .collection-title,.use-motion .comments,.use-motion .menu-item,.use-motion .motion-element,.use-motion .pagination,.use-motion .post-block,.use-motion .post-body,.use-motion .post-header{opacity:initial}.use-motion .logo,.use-motion .site-subtitle,.use-motion .site-title{opacity:initial;top:initial}.use-motion .logo-line-before i{left:initial}.use-motion .logo-line-after i{right:initial}</style></noscript><link rel="stylesheet" type="text/css" href="/live2d/waifu.css"></head><body itemscope itemtype="http://schema.org/WebPage" lang="zh-CN"><div class="container sidebar-position-right page-post-detail"><div class="headband"></div><header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader"><div class="header-inner"><div class="site-brand-wrapper"><div class="site-meta"><div class="custom-logo-site-title"><a href="/" class="brand" rel="start"><span class="logo-line-before"><i></i></span> <span class="site-title">潘高的小站</span> <span class="logo-line-after"><i></i></span></a></div><h1 class="site-subtitle" itemprop="description">IT套路深，必须瞎折腾</h1></div><div class="site-nav-toggle"><button aria-label="切换导航栏"><span class="btn-bar"></span> <span class="btn-bar"></span> <span class="btn-bar"></span></button></div></div><nav class="site-nav"><ul id="menu" class="menu"><li class="menu-item menu-item-home"><a href="/" rel="section"><i class="menu-item-icon fa fa-fw fa-home"></i><br>首页</a></li><li class="menu-item menu-item-top"><a href="/top/" rel="section"><i class="menu-item-icon fa fa-fw fa-signal"></i><br>阅读榜</a></li><li class="menu-item menu-item-tags"><a href="/tags/" rel="section"><i class="menu-item-icon fa fa-fw fa-tags"></i><br>标签<span class="badge">21</span></a></li><li class="menu-item menu-item-categories"><a href="/categories/" rel="section"><i class="menu-item-icon fa fa-fw fa-th"></i><br>分类<span class="badge">11</span></a></li><li class="menu-item menu-item-archives"><a href="/archives/" rel="section"><i class="menu-item-icon fa fa-fw fa-align-left"></i><br>时间轴<span class="badge">53</span></a></li><li class="menu-item menu-item-about"><a href="/about/" rel="section"><i class="menu-item-icon fa fa-fw fa-user"></i><br>关于</a></li><li class="menu-item menu-item-search"><a href="javascript:;" class="popup-trigger"><i class="menu-item-icon fa fa-search fa-fw"></i><br>搜索</a></li></ul><div class="site-search"><div class="popup search-popup local-search-popup"><div class="local-search-header clearfix"><span class="search-icon"><i class="fa fa-search"></i> </span><span class="popup-btn-close"><i class="fa fa-times-circle"></i></span><div class="local-search-input-wrapper"><input autocomplete="off" placeholder="搜索..." spellcheck="false" type="text" id="local-search-input"></div></div><div id="local-search-result"></div></div></div></nav></div></header><main id="main" class="main"><div class="main-inner"><div class="content-wrap"><div id="content" class="content"><div id="posts" class="posts-expand"><div class="reading-progress-bar"></div><article class="post post-type-normal" itemscope itemtype="http://schema.org/Article"><div class="post-block"><link itemprop="mainEntityOfPage" href="https://blog.pangao.vip/常用生物信息学格式介绍/"><span hidden itemprop="author" itemscope itemtype="http://schema.org/Person"><meta itemprop="name" content="潘高"><meta itemprop="description" content="Just do it !<br>(翻译：我只是个搞 IT的！哭笑脸)"><meta itemprop="image" content="/images/avatar.jpg"></span><span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization"><meta itemprop="name" content="潘高的小站"></span><header class="post-header"><h2 class="post-title" itemprop="name headline">常用生物信息学格式介绍</h2><div class="post-meta"><span class="post-time"><span class="post-meta-item-icon"><i class="fa fa-calendar-o"></i> </span><span class="post-meta-item-text">发表于</span> <time title="创建时间：2018-01-05 23:41:22" itemprop="dateCreated datePublished" datetime="2018-01-05T23:41:22+08:00">2018-01-05</time> <span class="post-meta-divider">|</span> <span class="post-meta-item-icon"><i class="fa fa-calendar-check-o"></i> </span><span class="post-meta-item-text">更新于</span> <time title="修改时间：2019-03-07 23:41:49" itemprop="dateModified" datetime="2019-03-07T23:41:49+08:00">2019-03-07</time> </span><span class="post-category"><span class="post-meta-divider">|</span> <span class="post-meta-item-icon"><i class="fa fa-folder-o"></i> </span><span class="post-meta-item-text">分类于</span> <span itemprop="about" itemscope itemtype="http://schema.org/Thing"><a href="/categories/生物信息分析/" itemprop="url" rel="index"><span itemprop="name">生物信息分析</span></a></span> </span><span class="post-comments-count"><span class="post-meta-divider">|</span> <span class="post-meta-item-icon"><i class="fa fa-comment-o"></i> </span><span class="post-meta-item-text">评论数：</span> <a href="/常用生物信息学格式介绍/#comments" itemprop="discussionUrl"><span class="post-comments-count valine-comment-count" data-xid="/常用生物信息学格式介绍/" itemprop="commentCount"></span> </a></span><span id="/常用生物信息学格式介绍/" class="leancloud_visitors" data-flag-title="常用生物信息学格式介绍"><span class="post-meta-divider">|</span> <span class="post-meta-item-icon"><i class="fa fa-eye"></i> </span><span class="post-meta-item-text">热度：</span> <span class="leancloud-visitors-count"></span> <span>℃</span></span><div class="post-symbolscount"><span class="post-meta-item-icon"><i class="fa fa-file-word-o"></i> </span><span class="post-meta-item-text">本文字数：</span> <span title="本文字数">14k</span> <span class="post-meta-divider">|</span> <span class="post-meta-item-icon"><i class="fa fa-clock-o"></i> </span><span class="post-meta-item-text">阅读时长 &asymp;</span> <span title="阅读时长">23 分钟</span></div></div></header><div class="post-body" itemprop="articleBody"><h2 id="fasta"><a href="#fasta" class="headerlink" title="fasta"></a>fasta</h2><p><a href="http://genetics.bwh.harvard.edu/pph/FASTA.html" target="_blank" rel="noopener">fasta</a> （<a href="https://en.wikipedia.org/wiki/FASTA_format" target="_blank" rel="noopener">维基百科</a>）格式是最基本的表示序列信息（核苷酸或者蛋白质）的格式。这里简单介绍下，fasta格式的文件通常后缀名为 .fasta 或者 .fa ，其实这都无所谓，因为都是文本文件。 fasta 格式文件（可以包含多条序列）中的一条序列的通常表示方法如下：</p><a id="more"></a><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">&gt;gi|129295|sp|P01013|OVAX_CHICK GENE X PROTEIN (OVALBUMIN-RELATED)</span><br><span class="line">QIKDLLVSSSTDLDTTLVLVNAIYFKGMWKTAFNAEDTREMPFHVTKQESKPVQMMCMNNSFNVATLPAE</span><br><span class="line">KMKILELPFASGDLSMLVLLPDEVSDLERIEKTINFEKLTEWTNPNTMEKRRVKVYLPQMKIEEKYNLTS</span><br><span class="line">VLMALGMTDLFIPSANLTGISSAESLKISQAVHGAFMELSEDGIEMAGSTGVIEDIKHSPESEQFRADHP</span><br><span class="line">FLFLIKHNPTNTIVYFGRYWSP</span><br></pre></td></tr></table></figure><p>其中主要分为两个部分：</p><ul><li>第一部分是序列的定义行（单行），该行的开头是&gt;符号，紧跟着后面的就是该条序列的名称（具有唯一性，即不能和其它序列同名称），即&gt;号和后面的名称的第一字符间是没有任何空白的。一般第一个空格后面的内容即为可选的描述信息。如上面， gi|129295|sp|P01013|OVAX_CHICK为序列名称， 而GENE X PROTEIN (OVALBUMIN-RELATED)则为描述信息。注意：有点软件是把一整行当做名称的，所以在出现错误的时候可以查看下格式是否正确。</li><li>第二部分就是序列，所有的序列碱基或者氨基酸可以都放在一行存储，也可以多行存储，但是建议大家多行存储且单行长度不超过80个字符，因为这样容易阅读。且序列的多行之间不能有空行，序列信息描述的第一行与序列数据的第一行之间不能有空行。其中序列数据主要是按照密码表来表示的，*表示是蛋白质翻译的结束。</li></ul><p>多行序列举例如下：<br></p><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">&gt;SEQUENCE_1</span><br><span class="line">MTEITAAMVKELRESTGAGMMDCKNALSETNGDFDKAVQLLREKGLGKAAKKADRLAAEG</span><br><span class="line">LVSVKVSDDFTIAAMRPSYLSYEDLDMTFVENEYKALVAELEKENEERRRLKDPNKPEHK</span><br><span class="line">IPQFASRKQLSDAILKEAEEKIKEELKAQGKPEKIWDNIIPGKMNSFIADNSQLDSKLTL</span><br><span class="line">MGQFYVMDDKKTVEQVIAEKEKEFGGKIKIVEFICFEVGEGLEKKTEDFAAEVAAQL</span><br><span class="line">&gt;SEQUENCE_2</span><br><span class="line">SATVSEINSETDFVAKNDQFIALTKDTTAHIQSNSLQSVEELHSSTINGVKFEEYLKSQI</span><br><span class="line">ATIGENLVVRRFATLKAGANGVVNGYIHTNGRVGVVIAAACDSAEVASKSRDLLRQICMH</span><br></pre></td></tr></table></figure><p></p><h2 id="fastq"><a href="#fastq" class="headerlink" title="fastq"></a>fastq</h2><p><a href="https://en.wikipedia.org/wiki/FASTQ_format" target="_blank" rel="noopener">fastq</a> 同样是以文本形式来存储序列信息的格式，后缀名通常为 .fastq 或者 .fq ，但是与 fasta 不相同的是，它除了存储序列本身外还存储了序列中每个单元所对应的质量分数，所以 fastq 格式通常用于高通量测试数据的存储。早期是有 Sanger 机构开发的，但是现在已经演变成一个高通量测序的标准了。</p><p>fastq 格式文件中一个完整的单元分为四行，每行的含义如下：<br>第一行： 以@开头，内容同fasta的描述行类似<br>第二行：具体的碱基序列<br>第三行：以+开头，后面的内容可以和第一行类似，也什么都没有只留+<br>第四行：以ASCII字符集（分数）编码来表示对应碱基的测序质量</p><p>比如下面的这个例子：<br></p><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">@SEQ_ID</span><br><span class="line">GATTTGGGGTTCAAAGCAGTATCGATCAAATAGTAAATCCATTTGTTCAACTCACAGTTT</span><br><span class="line">+</span><br><span class="line">!&apos;&apos;*((((***+))%%%++)(%%%%).1***-+*&apos;&apos;))**55CCF&gt;&gt;&gt;&gt;&gt;&gt;CCCCCCC65</span><br></pre></td></tr></table></figure><p></p><p>下面以Illumina和NCBI SRA两个测序数据来源来讲讲它们之间的区别：<br>通常我们获取测序数据有两种途径，一种是自己通过仪器测定，一种是在公共数据库中(比如之前说到的 NCBI 中的 SRA 数据库)获取，这两种方式主要是在序列名称的命名上和测序质量表示方式上有所不同。</p><p>Illumina 序列名称：<br></p><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">@HWUSI-EAS100R:6:73:941:1973#0/1</span><br></pre></td></tr></table></figure><p></p><p>上述以：隔开的每个字段的含义如下：</p><table><thead><tr><th>HWUSI-EAS100R</th><th>the unique instrument name</th></tr></thead><tbody><tr><td>6</td><td>flowcell lane</td></tr><tr><td>3</td><td>tile number within the flowcell lane</td></tr><tr><td>941</td><td>‘x’-coordinate of the cluster within the tile</td></tr><tr><td>1973</td><td>‘y’-coordinate of the cluster within the tile</td></tr><tr><td>#0</td><td>index number for a multiplexed sample (0 for no indexing)</td></tr><tr><td>/1</td><td>the member of a pair, /1 or /2 (paired-end or mate-pair reads only)</td></tr></tbody></table><h2 id="NCBI-SRA-数据库"><a href="#NCBI-SRA-数据库" class="headerlink" title="NCBI SRA 数据库"></a>NCBI SRA 数据库</h2><p>将测序数据提交到 NCBI 的 SRA 数据库时， SRA 数据库会为每一个样本提供一个编号，一般是 SRRxxxxx ，所以从 SRA 数据库上下载公共的测试数据（原始格式为 .sra ， 需特定工具转换为 fastq ），其 fastq 格式文件中每个单元的名称是以 SRA 编号接数字加以区分的。比如下面的这个示例：<br></p><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">@SRR001666.1 071112_SLXA-EAS1_s_7:5:1:817:345 length=36</span><br><span class="line">GGGTGATGGCCGCTGCCGATGGCGTCAAATCCCACC</span><br><span class="line">+SRR001666.1 071112_SLXA-EAS1_s_7:5:1:817:345 length=36</span><br><span class="line">IIIIIIIIIIIIIIIIIIIIIIIIIIIIII9IG9IC</span><br></pre></td></tr></table></figure><p></p><blockquote><p>需要注意的是：当把测序数据上传到 SRA 数据库时，它通常会将表示质量的分数 转换为标准的 Sanger 格式 。</p></blockquote><h5 id="质量分数表示法："><a href="#质量分数表示法：" class="headerlink" title="质量分数表示法："></a>质量分数表示法：</h5><p>由于测序仪器的不同等因素所以对碱基测序质量的表示方式也不相同，在 Fastq 格式文件中，用ASCII码表来表示每个碱基的测序质量，下面介绍几种不同的方案：<br><img src="https://ws4.sinaimg.cn/large/006tKfTcgy1g0umxr08r8j30sg0fggqo.jpg" alt="image"><br>其中有五种表示方法，Sanger 的码表范围为 ! 至 I ，其对应的数值为 33-73，如果减去 33（即 Phred + 33 表示法）这个基数则范围转换为 0-40，即如果某一个碱基的测序质量为 ! 则对应的测序质量分数为 0，表示测序质量低。其它几种表示法类似（X,I,J,L）。这里介绍测序质量的表示方法是因为后面有的软件是要指定测序数据的质量表示方法。</p><h2 id="gff2"><a href="#gff2" class="headerlink" title="gff2"></a>gff2</h2><p><a href="http://gmod.org/wiki/GFF2" target="_blank" rel="noopener">GFF</a>(General Feature Format)是一种用于描述基因或者其它序列元素的文件格式，GFF 有几个版本，早期的第 Version 2 和现在的 Version 3. Version 2 是由 Sanger 机构所制定的，而 Version 3 是由 Sequence Ontology Project 制定。正是由于有统一的格式来表示基因等元素，使得 GFF 格式的文件被广泛的使用与 mapping 与基因组数据可视化方面。<br>GFF2 文件格式是由 tab 隔开的九列值，每一行的九个字段的含义如下：<br></p><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">Chr1  curated  CDS 365647  365963  .  +  1  Transcript &quot;R119.7&quot;</span><br></pre></td></tr></table></figure><p></p><p>第一列： reference sequence， 该列表示的是特征元素所在的染色体（或者scaffold，或者contig），也就是在基因组中的坐标系统，后续一切的注释信息都是基于此列。<br>第二列：source，该列表示改行注释信息的来源，比如上述的一行表示该行的CDS注释信息来自名为“curated”的注释。<br>第三列：feature，或者说是method，type， 表示的是该注释的类型，比如上述表示改行注释为CDS信息，可以将source和feature结合起来描述的更加详细。<br>第四列：start position，在reference sequence上的开始位置（坐标），通常是从1为起点而不是0。<br>第五列：end position， 在reference sequence上的结束位置（坐标），一般是大于start position的。<br>第六列：score， 表示该行feature的分数，比如序列相似性等，如果没有对应的分数可以用.代替。<br>第七列：strand，feature所在链，+表示正链，-表示负链，.表示不确定或者与链无关。<br>第八列：phase，与蛋白质编码相关，一般是用于CDS，值的范围为0-2，表示编码时阅读框的移动相位。<br>下面这段描述很详细:<br>‘0’ indicates that the specified region is in frame, i.e. that its first base corresponds to the first base of a codon. ‘1’ indicates that there is one extra base, i.e. that the second base of the region corresponds to the first base of a codon, and ‘2’ means that the third base of the region is the first base of a codon. If the strand is ‘-‘, then the first base of the region is value of<end>, because the corresponding coding region will run from<end>to<start>on the reverse strand.<br>第九列：group，或者称为attributes，是用于对改行注释更多的描述，以键值对的形式，比如上面的例子表示该CDS是属于名为R119.7的transcript。该列中可以存在多个属性，属性之间是用;隔开的。<br>对于GFF格式的理解主要是集中在最后一列，有以下集中情况：</start></end></end></p><ol><li><p>对于单个feature</p><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">Chr3   giemsa heterochromatin  4500000 6000000 . . .   Band 3q12.1</span><br></pre></td></tr></table></figure></li><li><p>对于属于同一集合的多个feature</p><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">IV     curated exon    5506900 5506996 . + .   Transcript B0273.1</span><br><span class="line">IV     curated exon    5506026 5506382 . + .   Transcript B0273.1</span><br><span class="line">IV     curated exon    5506558 5506660 . + .   Transcript B0273.1</span><br><span class="line">IV     curated exon    5506738 5506852 . + .   Transcript B0273.1</span><br></pre></td></tr></table></figure></li></ol><p>比如上面这个例子就表示这四个exon都是属于同一个名为B0273.1的transcript，这是表示一个完整transcript结构的最基本要求。<br>GFF2还可用于序列比对结果表示等其它方面，这里不做介绍了。</p><h2 id="gtf-gff2-5"><a href="#gtf-gff2-5" class="headerlink" title="gtf(gff2.5)"></a>gtf(gff2.5)</h2><p><a href="http://mblab.wustl.edu/GTF2.html" target="_blank" rel="noopener">GTF</a>（Gene Transfer Format）格式是借鉴于GFF2格式，也被称为GFF2.5，大部分字段的定义是和GFF2相同的，只是每行的第九列必须带有如下四个域，具体为gene_id value; transcript_id value; 这样的设计是为了适应一个基因的多个转录本这种情况。比如下面的这个例子：<br></p><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">AB000123    Twinscan     CDS    193817    194022    .    -    2    gene_id &quot;AB000123.1&quot;; transcript_id &quot;AB00123.1.2&quot;;</span><br><span class="line">AB000123    Twinscan     CDS    199645    199752    .    -    2    gene_id &quot;AB000123.1&quot;; transcript_id &quot;AB00123.1.2&quot;;</span><br><span class="line">AB000123    Twinscan     CDS    200369    200508    .    -    1    gene_id &quot;AB000123.1&quot;; transcript_id &quot;AB00123.1.2&quot;;</span><br><span class="line">AB000123    Twinscan     CDS    215991    216028    .    -    0    gene_id &quot;AB000123.1&quot;; transcript_id &quot;AB00123.1.2&quot;;</span><br><span class="line">AB000123    Twinscan     start_codon   216026    216028    .    -    .    gene_id    &quot;AB000123.1&quot;; transcript_id &quot;AB00123.1.2&quot;;</span><br><span class="line">AB000123    Twinscan     stop_codon    193814    193816    .    -    .    gene_id    &quot;AB000123.1&quot;; transcript_id &quot;AB00123.1.2&quot;;</span><br></pre></td></tr></table></figure><p></p><h2 id="gff3"><a href="#gff3" class="headerlink" title="gff3"></a>gff3</h2><p><a href="http://www.sequenceontology.org/gff3.shtml" target="_blank" rel="noopener">GFF2</a>（<a href="http://gmod.org/wiki/GFF3" target="_blank" rel="noopener">维基百科</a>）格式早期用的比较多，但是现在用的多的是GFF3格式，这也是好多软件所支持的，比如Gbrowse， Jbrowse等基因组数据可视化工具。<br>先看下面这个简单的例子：<br></p><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">##gff-version 3</span><br><span class="line">ctg123  .  exon  1300  1500  .  +  .  ID=exon00001</span><br><span class="line">ctg123  .  exon  1050  1500  .  +  .  ID=exon00002</span><br><span class="line">ctg123  .  exon  3000  3902  .  +  .  ID=exon00003</span><br><span class="line">ctg123  .  exon  5000  5500  .  +  .  ID=exon00004</span><br><span class="line">ctg123  .  exon  7000  9000  .  +  .  ID=exon00005</span><br></pre></td></tr></table></figure><p></p><p>第一行的##gff-version 3通常是需要的，而且必须是在文件的第一行。<br>前八列和GFF2、GFF2.5类似，但是有几点是要特别注意的，主要是将GFF3注释数据用于基因组浏览器时，字段中的一些特殊字符比如空格，&gt; %等都需要使用URL编码进行转换才能准确的在web中进行展示。<br>第九列同样是表示attributes，采用的同样是键值对的形式（tag=value），只是这里有几个特定的键，具体如下：<br>ID，feature在整个GFF3文件中唯一的标识符；<br>Name，feature的名字，不同于ID，Name不要求唯一，只是方便用户浏览；<br>Alias， 相当于feature的别名；<br>Parent，表明该feature所属的上一级feature 的ID，这种关系可用于exons-transcripts，transcripts-genes，可以看出一个feature可以拥有多个子feature；<br>Target， 主要是用于序列比对结果的展示，value的格式为target_id start end [strand], 其中如果target_id中含有空格则需转换为%20；<br>后面还有些其它属性比如Note等，这里不再做详细描述。<br>下面再来看下典型的例子：</p><ul><li>蛋白质编码基因结构<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">ctg123 example gene            1050 9000 . + . ID=EDEN;Name=EDEN;Note=protein kinase</span><br><span class="line"></span><br><span class="line">ctg123 example mRNA            1050 9000 . + . ID=EDEN.1;Parent=EDEN;Name=EDEN.1;Index=1</span><br><span class="line">ctg123 example five_prime_UTR  1050 1200 . + . Parent=EDEN.1</span><br><span class="line">ctg123 example CDS             1201 1500 . + 0 Parent=EDEN.1</span><br><span class="line">ctg123 example CDS             3000 3902 . + 0 Parent=EDEN.1</span><br><span class="line">ctg123 example CDS             5000 5500 . + 0 Parent=EDEN.1</span><br><span class="line">ctg123 example CDS             7000 7608 . + 0 Parent=EDEN.1</span><br><span class="line">ctg123 example three_prime_UTR 7609 9000 . + . Parent=EDEN.1</span><br><span class="line"></span><br><span class="line">ctg123 example mRNA            1050 9000 . + . ID=EDEN.2;Parent=EDEN;Name=EDEN.2;Index=1</span><br><span class="line">ctg123 example five_prime_UTR  1050 1200 . + . Parent=EDEN.2</span><br><span class="line">ctg123 example CDS             1201 1500 . + 0 Parent=EDEN.2</span><br><span class="line">ctg123 example CDS             5000 5500 . + 0 Parent=EDEN.2</span><br><span class="line">ctg123 example CDS             7000 7608 . + 0 Parent=EDEN.2</span><br><span class="line">ctg123 example three_prime_UTR 7609 9000 . + . Parent=EDEN.2</span><br><span class="line"></span><br><span class="line">ctg123 example mRNA            1300 9000 . + . ID=EDEN.3;Parent=EDEN;Name=EDEN.3;Index=1</span><br><span class="line">ctg123 example five_prime_UTR  1300 1500 . + . Parent=EDEN.3</span><br><span class="line">ctg123 example five_prime_UTR  3000 3300 . + . Parent=EDEN.3</span><br><span class="line">ctg123 example CDS             3301 3902 . + 0 Parent=EDEN.3</span><br><span class="line">ctg123 example CDS             5000 5500 . + 1 Parent=EDEN.3</span><br><span class="line">ctg123 example CDS             7000 7600 . + 1 Parent=EDEN.3</span><br><span class="line">ctg123 example three_prime_UTR 7601 9000 . + . Parent=EDEN.3</span><br></pre></td></tr></table></figure></li></ul><p>一个名为EDEN的基因拥有三个转录本，分别名为EDEN.1 EDEN.2 EDEN.3， 每个转录本又有UTR和CDS等信息。</p><ul><li><p>序列比对</p><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">ctg123 est EST_match 1050 1500 . + . ID=Match1;Name=agt830.5;Target=agt830.5 1 451</span><br><span class="line">ctg123 est EST_match 3000 3202 . + . ID=Match1;Name=agt830.5;Target=agt830.5 452 654</span><br><span class="line"></span><br><span class="line">ctg123 est EST_match 5410 5500 . - . ID=Match2;Name=agt830.3;Target=agt830.3 505 595</span><br><span class="line">ctg123 est EST_match 7000 7503 . - . ID=Match2;Name=agt830.3;Target=agt830.3 1 504</span><br><span class="line"></span><br><span class="line">ctg123 est EST_match 1050 1500 . + . ID=Match3;Name=agt221.5;Target=agt221.5 1 451</span><br><span class="line">ctg123 est EST_match 5000 5500 . + . ID=Match3;Name=agt221.5;Target=agt221.5 452 952</span><br><span class="line">ctg123 est EST_match 7000 7300 . + . ID=Match3;Name=agt221.5;Target=agt221.5 953 1253</span><br></pre></td></tr></table></figure></li><li><p>定量数据</p><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">ctg123 affy microarray_oligo   1 100 281 . . Name=Expt1</span><br><span class="line">ctg123 affy microarray_oligo 101 200 183 . . Name=Expt1</span><br><span class="line">ctg123 affy microarray_oligo 201 300 213 . . Name=Expt1</span><br><span class="line">ctg123 affy microarray_oligo 301 400 191 . . Name=Expt1</span><br><span class="line">ctg123 affy microarray_oligo 401 500 288 . . Name=Expt1</span><br><span class="line">ctg123 affy microarray_oligo 501 600 184 . . Name=Expt1</span><br></pre></td></tr></table></figure></li><li><p>含Fasta格式的GFF3格式文件</p><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">##gff-version 3</span><br><span class="line">ctg123 . exon            1300  1500  .  +  .  ID=exon00001</span><br><span class="line">ctg123 . exon            1050  1500  .  +  .  ID=exon00002</span><br><span class="line">ctg123 . exon            3000  3902  .  +  .  ID=exon00003</span><br><span class="line">ctg123 . exon            5000  5500  .  +  .  ID=exon00004</span><br><span class="line">ctg123 . exon            7000  9000  .  +  .  ID=exon00005</span><br><span class="line">##FASTA</span><br><span class="line">&gt;ctg123</span><br><span class="line">cttctgggcgtacccgattctcggagaacttgccgcaccattccgccttg</span><br><span class="line">tgttcattgctgcctgcatgttcattgtctacctcggctacgtgtggcta</span><br><span class="line">tctttcctcggtgccctcgtgcacggagtcgagaaaccaaagaacaaaaa</span><br><span class="line">aagaaattaaaatatttattttgctgtggtttttgatgtgtgttttttat</span><br><span class="line">aatgatttttgatgtgaccaattgtacttttcctttaaatgaaatgtaat</span><br><span class="line">cttaaatgtatttccgacgaattcgaggcctgaaaagtgtgacgccattc</span><br><span class="line">...</span><br></pre></td></tr></table></figure></li></ul><p>该GFF3文件中含有对应的序列，以##FASTA作为标示。</p><h2 id="bed"><a href="#bed" class="headerlink" title="bed"></a>bed</h2><p>bed（<a href="http://www.genome.ucsc.edu/FAQ/FAQformat.html#format1" target="_blank" rel="noopener">genome介绍</a>、<a href="http://bedtools.readthedocs.org/en/latest/content/general-usage.html" target="_blank" rel="noopener">bedtools介绍</a>、 <a href="http://asia.ensembl.org/info/website/upload/bed.html?redirect=no" target="_blank" rel="noopener">asia介绍</a>）格式同样是用于展示序列注释信息，有相应的软件来处理这类格式的文件，如bedtools。可以用在类似GBrowse这样的基因组数据可视化工具中。 以tab隔开，它必须的三个字段为 chrom、chromStart、chromEnd，还有9个可选字段。<br>注意：用于在GBrowse上展示相关注释的bed格式通常第一行有一个关于track的描述信息。<br>比如下面的例子：<br></p><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">track name=pairedReads description=&quot;Clone Paired Reads&quot; useScore=1</span><br><span class="line">chr22 1000 5000 cloneA 960 + 1000 5000 0 2 567,488, 0,3512</span><br><span class="line">chr22 2000 6000 cloneB 900 - 2000 6000 0 2 433,399, 0,3601</span><br></pre></td></tr></table></figure><p></p><h2 id="sam-bam"><a href="#sam-bam" class="headerlink" title="sam/bam"></a>sam/bam</h2><p><a href="http://blog.sina.com.cn/s/blog_670445240101l30k.html" target="_blank" rel="noopener">http://blog.sina.com.cn/s/blog_670445240101l30k.html</a><br><a href="http://genome.sph.umich.edu/wiki/SAM" target="_blank" rel="noopener">http://genome.sph.umich.edu/wiki/SAM</a><br><a href="https://samtools.github.io/hts-specs/SAMv1.pdf" target="_blank" rel="noopener">https://samtools.github.io/hts-specs/SAMv1.pdf</a><br>在生物信息学中尤其是高通量测序数据分析中，大部分的操作都是在实现短片段序列与参考序列的比对（mapping），比如bowtie等，这就涉及到如何使用一个统一的格式来表示这种mapping结果呢，sam（Sequence Alignment/Map）格式就是来解决这个问题的。sam文件拥有头部描述和详细比对两部分，其中头部描述是以@开头，后面紧跟两个缩写字母表示相应的含义，SAM分为两部分，注释信息（header section）和比对结果部分（alignment section），注释信息可有可无，都是以@开头，用不同的tag表示不同的信息，主要有@HD，说明符合标准的版本、对比序列的排列顺序；@SQ，参考序列说明；@RG，比对上的序列（read）说明；@PG，使用的程序说明；@CO，任意的说明信息。而详细比对部分是通过11个tab隔开的字段来表示。<br>下面主要讲解下详细比对部分字段的具体含义：<br><img src="https://ws3.sinaimg.cn/large/006tKfTcgy1g0ung2b9yhj30g005hwfa.jpg" alt="image"><br>其中：</p><ol><li>QNAME 表示的是查询序列的名称即短片段（reads）的名称；</li><li>FLAG 以整数来表示比对的结果，不同数值有不同的意义，数值也可以是下列数的组合；<br><img src="https://ws2.sinaimg.cn/large/006tKfTcgy1g0ungd9nx4j30ec05wt8v.jpg" alt="image"></li></ol><p>比如如果FLAG是4的话则表示该reads没有比对到参考序列上，flag为16表示single-end reads比对到参考序列的反链上，<br>flag为83（64+16+2+1）表示paired-end reads中的第一个reads比对到参考序列上了。</p><ol start="3"><li>RNAME 表示参考序列的名称，比如基因组的染色体编号等，如果没有比对上则显示为*；</li><li>POS 表示比对的起始位置，以1开始计数，如果没有比对上则显示为0；</li><li>MAPQ 比对质量；</li><li>CIGAR CIGAR 字符串，即比对的详细情况，简要比对信息表达式（Compact Idiosyncratic Gapped Alignment Report），其以参考序列为基础，使用数字加字母表示比对结果，比如3S6M1P1I4M，前三个碱基被剪切去除了，然后6个比对上了，然后打开了一 个缺口，有一个碱基插入，最后是4个比对上了，是按照顺序的；</li><li>RNEXT 双末端测序中下一个reads比对的参考系列的名称，如果没有则用*表示，如果和前一个reads比对到同一个参考序列则用=表示；</li><li>PNEXT 下一个reads比对到参考序列上的位置，如果没有则用0表示；</li><li>ISIZE/TLEN query序列的模板长度或者插入长度，Template的长度，最左边得为正，最右边的为负，中间的不用定义正负，不分区段（single-segment)的比对上，或者不可用时，此处为0；</li><li>reads的序列信息；</li><li>reads的序列质量信息，同FASTQ。<br>后面还有些可选字段，比如：<br>可选字段（optional fields)，格式如：TAG:TYPE:VALUE，其中TAG有两个大写字母组成，每个TAG代表一类信息，每一行一个TAG只能出现一次，TYPE表示TAG对应值的类型，可以是字符串、整数、字节、数组等。<br>示例：<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">:497:R:-272+13M17D24M	113	1	497	37	37M	15	100338662	0	CGGGTCTGACCTGAGGAGAACTGTGCTCCGCCTTCAG	0;==-==9;&gt;&gt;&gt;&gt;&gt;=&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;=&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;	XT:A:U	NM:i:0	SM:i:37	AM:i:0	X0:i:1	X1:i:0	XM:i:0	XO:i:0	XG:i:0	MD:Z:37</span><br><span class="line">:20389:F:275+18M2D19M	99	1	17644	0	37M	=	17919	314	TATGACTGCTAATAATACCTACACATGTTAGAACCAT	&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&gt;&lt;&lt;&gt;&gt;&gt;&lt;&lt;&gt;&gt;4::&gt;&gt;:&lt;9	RG:Z:UM0098:1	XT:A:R	NM:i:0	SM:i:0	AM:i:0	X0:i:4	X1:i:0	XM:i:0	XO:i:0	XG:i:0	MD:Z:37</span><br><span class="line">:20389:F:275+18M2D19M	147	1	17919	0	18M2D19M	=	17644	-314	GTAGTACCAACTGTAAGTCCTTATCTTCATACTTTGT	;44999;499&lt;8&lt;8&lt;&lt;&lt;8&lt;&lt;&gt;&lt;&lt;&lt;&lt;&gt;&lt;7&lt;;&lt;&lt;&lt;&gt;&gt;&lt;&lt;	XT:A:R	NM:i:2	SM:i:0	AM:i:0	X0:i:4	X1:i:0	XM:i:0	XO:i:1	XG:i:2	MD:Z:18^CA19</span><br><span class="line">:21597+10M2I25M:R:-209	83	1	21678	0	8M2I27M	=	21469	-244	CACCACATCACATATACCAAGCCTGGCTGTGTCTTCT	&lt;;9&lt;&lt;5&gt;&lt;&lt;&lt;&lt;&gt;&lt;&lt;&lt;&gt;&gt;&lt;&lt;&gt;&lt;&gt;&gt;&lt;9&gt;&gt;&lt;&gt;&gt;&gt;9&gt;&gt;&gt;&lt;&gt;	XT:A:R	NM:i:2	SM:i:0	AM:i:0	X0:i:5	X1:i:0	XM:i:0	XO:i:1	XG:i:2	MD:Z:35</span><br></pre></td></tr></table></figure></li></ol><p>解释：</p><p>其中可以看出Aligenment 2 和 Alignment 3是成对的reads，其插入长度为314。<br>bam格式中的b是binary的意思，是sam格式的二进制表示方式，为什么要用二进制表示呢？ 因为sam格式文件大小通常是十分大的，一般是以G为单位，所以为了减少存储量等因素而将sam转换为二进制格式以便于分析。<br>sam/bam格式是由特定的一些软件（比如samtools）来处理的，包括格式互转、排序、建立索引、搜寻突变等操作，后续分析中会详细讲解samtools工具的使用方法。<br><img src="https://ws1.sinaimg.cn/large/006tKfTcgy1g0unj39xr5j30sg08e0ue.jpg" alt="image"></p><h2 id="vcf"><a href="#vcf" class="headerlink" title="vcf"></a>vcf</h2><p><a href="http://samtools.github.io/hts-specs/VCFv4.2.pdf" target="_blank" rel="noopener">vcf</a>（Variant Call Format）格式是用于表示突变信息的文本格式，可以用来表示single nucleotide variants, insertions/deletions, copy number variants and structural variants等。VCF格式同样是分为两大部分，一部分是注释描述信息，一部分是具体的突变信息，其中注释信息是以##开头的，我们来看下面这个例子：<br><img src="https://ws3.sinaimg.cn/large/006tKfTcgy1g0unkjrkwwj30s30ajgny.jpg" alt="image"><br>我们着重来关注第二部分的每列字段是什么含义：<br>CHROM 即chromosome， 染色体名称；<br>POS 即position， 发生突变的参考序列的位置（从1开始计数）；<br>ID 突变的名称；<br>REF 参考序列POS上的碱基；<br>ALT 发生突变的碱基，多个的话以,连接， 可选符号为ATCGN*，大小写敏感；<br>QUAL 基于Phred格式的表示ALT的质量，也可以理解为可靠性；<br>FILTER 过滤后的状态，即按照可靠性进行筛选；<br>INFO 额外信息，可结合注释描述信息进行理解<br><img src="https://ws4.sinaimg.cn/large/006tKfTcgy1g0unlikupmj30nh0ea760.jpg" alt="image"><br>针对vcf格式有如bcftools等软件进行处理。</p><hr></div><script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script><ins class="adsbygoogle" style="display:block;text-align:center" data-ad-layout="in-article" data-ad-format="fluid" data-ad-client="ca-pub-5317369003185250" data-ad-slot="2967493559"></ins><script>(adsbygoogle=window.adsbygoogle||[]).push({})</script><div class="popular-posts-header">相关文章推荐</div><ul class="popular-posts"><li class="popular-posts-item"><div class="popular-posts-title"><a href="/外显子、内含子、mRNA、CDS、ORF的区别与联系/" rel="bookmark">外显子、内含子、mRNA、CDS、ORF的区别与联系</a></div></li><li class="popular-posts-item"><div class="popular-posts-title"><a href="/生物信息分析中的reads是什么/" rel="bookmark">生物信息分析中的reads是什么</a></div></li><li class="popular-posts-item"><div class="popular-posts-title"><a href="/生物信息分析中的测序概念/" rel="bookmark">生物信息分析中的测序概念</a></div></li></ul><div><div><div style="text-align:center;color:#ccc;font-size:14px">-------------已经触及底线 <i class="fa fa-paw"></i> 感谢您的阅读-------------</div></div></div><div><div style="padding:10px 0;margin:20px auto;width:90%;text-align:center"><div>如果你觉得这篇文章对你有帮助，或者想给我微小的工作一点点资瓷，请随意打赏。</div><button id="rewardButton" disable="enable" onclick="var qr = document.getElementById(&quot;QR&quot;); if (qr.style.display === 'none') {qr.style.display='block';} else {qr.style.display='none'}"><span>赞赏</span></button><div id="QR" style="display:none"><div id="wechat" style="display:inline-block"><img id="wechat_qr" src="/images/wechatpay.jpg" alt="潘高 微信支付"><p>微信支付</p></div><div id="alipay" style="display:inline-block"><img id="alipay_qr" src="/images/alipay.png" alt="潘高 支付宝"><p>支付宝</p></div></div></div></div><div><div class="my_post_copyright"><script src="//cdn.bootcss.com/clipboard.js/1.5.10/clipboard.min.js"></script><script type="text/javascript" src="http://jslibs.wuxubj.cn/sweetalert_mini/jquery-1.7.1.min.js"></script><script src="http://jslibs.wuxubj.cn/sweetalert_mini/sweetalert.min.js"></script><link rel="stylesheet" type="text/css" href="http://jslibs.wuxubj.cn/sweetalert_mini/sweetalert.mini.css"><p><span>本文标题:</span>常用生物信息学格式介绍</p><p><span>文章作者:</span>潘高</p><p><span>发布时间:</span>2018年01月05日 - 23:41:22</p><p><span>最后更新:</span>2019年03月07日 - 23:41:49</p><p><span>原始链接:</span><a href="/常用生物信息学格式介绍/" title="常用生物信息学格式介绍">https://blog.pangao.vip/常用生物信息学格式介绍/</a> <span class="copy-path" title="点击复制文章链接"><i class="fa fa-clipboard" data-clipboard-text="https://blog.pangao.vip/常用生物信息学格式介绍/" aria-label="复制成功！"></i></span></p><p><span>许可协议:</span><i class="fa fa-creative-commons"></i> <a rel="license" href="https://creativecommons.org/licenses/by-nc-nd/4.0/" target="_blank" title="Attribution-NonCommercial-NoDerivatives 4.0 International (CC BY-NC-ND 4.0)">署名-非商业性使用-禁止演绎 4.0 国际</a> 转载请保留原文链接及作者。</p></div><script>var clipboard=new Clipboard(".fa-clipboard");clipboard.on("success",$(function(){$(".fa-clipboard").click(function(){swal({title:"",text:"复制成功",html:!1,timer:500,showConfirmButton:!1})})}))</script></div><footer class="post-footer"><div class="post-tags"><a href="/tags/生物信息分析/" rel="tag"><i class="fa fa-tag"></i> 生物信息分析</a></div><div class="post-widgets"><div class="wp_rating"><div id="wpac-rating"></div></div></div><div class="post-nav"><div class="post-nav-next post-nav-item"><a href="/生物信息分析中的测序概念/" rel="next" title="生物信息分析中的测序概念"><i class="fa fa-chevron-left"></i> 生物信息分析中的测序概念</a></div><span class="post-nav-divider"></span><div class="post-nav-prev post-nav-item"><a href="/Perl解压zip/" rel="prev" title="Perl解压zip">Perl解压zip <i class="fa fa-chevron-right"></i></a></div></div></footer></div></article></div></div><div></div><script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script><ins class="adsbygoogle" style="display:block" data-ad-client="ca-pub-5317369003185250" data-ad-slot="9014150860" data-ad-format="auto" data-full-width-responsive="true"></ins><script>(adsbygoogle=window.adsbygoogle||[]).push({})</script><div class="comments" id="comments"></div></div><div class="sidebar-toggle"><div class="sidebar-toggle-line-wrap"><span class="sidebar-toggle-line sidebar-toggle-line-first"></span> <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span> <span class="sidebar-toggle-line sidebar-toggle-line-last"></span></div></div><aside id="sidebar" class="sidebar"><div id="sidebar-dimmer"></div><div class="sidebar-inner"><ul class="sidebar-nav motion-element"><li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap">文章目录</li><li class="sidebar-nav-overview" data-target="site-overview-wrap">站点概览</li></ul><div class="site-overview-wrap sidebar-panel"><div class="site-overview"><div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person"><a href="/"><img class="site-author-image" itemprop="image" src="/images/avatar.jpg" alt="潘高"></a><p class="site-author-name" itemprop="name">潘高</p><p class="site-description motion-element" itemprop="description">Just do it !<br>(翻译：我只是个搞 IT的！哭笑脸)</p></div><nav class="site-state motion-element"><div class="site-state-item site-state-posts"><a href="/archives/"><span class="site-state-item-count">53</span> <span class="site-state-item-name">日志</span></a></div><div class="site-state-item site-state-categories"><a href="/categories/index.html"><span class="site-state-item-count">11</span> <span class="site-state-item-name">分类</span></a></div><div class="site-state-item site-state-tags"><a href="/tags/index.html"><span class="site-state-item-count">21</span> <span class="site-state-item-name">标签</span></a></div></nav><div id="days"></div><script>function show_date_time(){window.setTimeout("show_date_time()",1e3),BirthDay=new Date("12/22/2018 00:00:00"),today=new Date,timeold=today.getTime()-BirthDay.getTime(),sectimeold=timeold/1e3,secondsold=Math.floor(sectimeold),msPerDay=864e5,e_daysold=timeold/msPerDay,daysold=Math.floor(e_daysold),e_hrsold=24*(e_daysold-daysold),hrsold=setzero(Math.floor(e_hrsold)),e_minsold=60*(e_hrsold-hrsold),minsold=setzero(Math.floor(60*(e_hrsold-hrsold))),seconds=setzero(Math.floor(60*(e_minsold-minsold))),document.getElementById("days").innerHTML="已安全运行 "+daysold+" 天 "+hrsold+" 小时 "+minsold+" 分 "+seconds+" 秒"}function setzero(e){return 10>e&&(e="0"+e),e}show_date_time()</script></div></div><div class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active"><div class="post-toc"><div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#fasta"><span class="nav-number">1.</span> <span class="nav-text">fasta</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#fastq"><span class="nav-number">2.</span> <span class="nav-text">fastq</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#NCBI-SRA-数据库"><span class="nav-number">3.</span> <span class="nav-text">NCBI SRA 数据库</span></a><ol class="nav-child"><li class="nav-item nav-level-5"><a class="nav-link" href="#质量分数表示法："><span class="nav-number">3.0.0.1.</span> <span class="nav-text">质量分数表示法：</span></a></li></ol></li></ol><li class="nav-item nav-level-2"><a class="nav-link" href="#gff2"><span class="nav-number">4.</span> <span class="nav-text">gff2</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#gtf-gff2-5"><span class="nav-number">5.</span> <span class="nav-text">gtf(gff2.5)</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#gff3"><span class="nav-number">6.</span> <span class="nav-text">gff3</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#bed"><span class="nav-number">7.</span> <span class="nav-text">bed</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#sam-bam"><span class="nav-number">8.</span> <span class="nav-text">sam/bam</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#vcf"><span class="nav-number">9.</span> <span class="nav-text">vcf</span></a></li></div></div></div><div class="back-to-top"><i class="fa fa-arrow-up"></i> <span id="scrollpercent"><span>0</span>%</span></div></div></aside></div></main><footer id="footer" class="footer"><div class="footer-inner"><script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script><div class="copyright">&copy; 2018 – <span itemprop="copyrightYear">2019</span> <span class="with-love" id="animate"><i class="fa fa-heart"></i> </span><span class="author" itemprop="copyrightHolder">潘高</span></div><div class="powered-by"><i class="fa fa-user-md"></i> <span id="busuanzi_container_site_uv">站点访客数: <span id="busuanzi_value_site_pv"></span></span></div><span class="post-meta-divider">|</span> <span class="post-meta-item-icon"><i class="fa fa-area-chart"></i> </span><span class="post-meta-item-text">站点总字数：</span> <span title="站点总字数">82k</span> <span class="post-meta-divider">|</span> <span class="post-meta-item-icon"><i class="fa fa-coffee"></i> </span><span class="post-meta-item-text">站点阅读时长 &asymp;</span> <span title="站点阅读时长">2:17</span></div></footer></div><script>"[object Function]"!==Object.prototype.toString.call(window.Promise)&&(window.Promise=null)</script><script src="/lib/jquery/index.js?v=2.1.3"></script><script src="/lib/velocity/velocity.min.js?v=1.2.1"></script><script src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script><script src="/lib/reading_progress/reading_progress.js"></script><script src="/js/src/utils.js?v=7.0.0"></script><script src="/js/src/motion.js?v=7.0.0"></script><script src="/js/src/affix.js?v=7.0.0"></script><script src="/js/src/schemes/pisces.js?v=7.0.0"></script><script src="/js/src/scrollspy.js?v=7.0.0"></script><script src="/js/src/post-details.js?v=7.0.0"></script><script src="/js/src/bootstrap.js?v=7.0.0"></script><script src="//cdn1.lncld.net/static/js/3.11.1/av-min.js"></script><script src="//unpkg.com/valine/dist/Valine.min.js"></script><script>var GUEST=["nick","mail","link"],guest="nick,mail,link";guest=guest.split(",").filter(function(i){return GUEST.indexOf(i)>-1}),new Valine({el:"#comments",verify:!1,notify:!0,appId:"w5RAfQXjT6iFfBtGwiTfxQOP-gzGzoHsz",appKey:"h55viQ2cLIrigHjPHDfIS2ly",placeholder:"风过留痕，雁过留声，君过留言。（建议在上方留下昵称和邮箱等信息，以便收到回复通知）",avatar:"robohash",meta:guest,pageSize:"10",visitor:!1})</script><script>function proceedsearch(){$("body").append('<div class="search-popup-overlay local-search-pop-overlay"></div>').css("overflow","hidden"),$(".search-popup-overlay").click(onPopupClose),$(".popup").toggle();var e=$("#local-search-input");e.attr("autocapitalize","none"),e.attr("autocorrect","off"),e.focus()}var isfetched=!1,isXml=!0,search_path="search.xml";0===search_path.length?search_path="search.xml":/json$/i.test(search_path)&&(isXml=!1);var path="/"+search_path,onPopupClose=function(e){$(".popup").hide(),$("#local-search-input").val(""),$(".search-result-list").remove(),$("#no-result").remove(),$(".local-search-pop-overlay").remove(),$("body").css("overflow","")},searchFunc=function(e,t,o){"use strict";$("body").append('<div class="search-popup-overlay local-search-pop-overlay"><div id="search-loading-icon"><i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i></div></div>').css("overflow","hidden"),$("#search-loading-icon").css("margin","20% auto 0 auto").css("text-align","center"),$.ajax({url:e,dataType:isXml?"xml":"json",async:!0,success:function(e){isfetched=!0,$(".popup").detach().appendTo(".header-inner");var n=isXml?$("entry",e).map(function(){return{title:$("title",this).text(),content:$("content",this).text(),url:$("url",this).text()}}).get():e,r=document.getElementById(t),s=document.getElementById(o),a=function(){var e=r.value.trim().toLowerCase(),t=e.split(/[\s\-]+/);t.length>1&&t.push(e);var o=[];if(e.length>0&&n.forEach(function(n){function r(t,o,n,r){for(var s=r[r.length-1],a=s.position,i=s.word,l=[],h=0;a+i.length<=n&&0!=r.length;){i===e&&h++,l.push({position:a,length:i.length});var p=a+i.length;for(r.pop();0!=r.length&&(s=r[r.length-1],a=s.position,i=s.word,p>a);)r.pop()}return c+=h,{hits:l,start:o,end:n,searchTextCount:h}}function s(e,t){var o="",n=t.start;return t.hits.forEach(function(t){o+=e.substring(n,t.position);var r=t.position+t.length;o+='<b class="search-keyword">'+e.substring(t.position,r)+"</b>",n=r}),o+=e.substring(n,t.end)}var a=!1,i=0,c=0,l=n.title.trim(),h=l.toLowerCase(),p=n.content.trim().replace(/<[^>]+>/g,""),u=p.toLowerCase(),f=decodeURIComponent(n.url).replace(/\/{2,}/g,"/"),d=[],g=[];if(""!=l&&(t.forEach(function(e){function t(e,t,o){var n=e.length;if(0===n)return[];var r=0,s=[],a=[];for(o||(t=t.toLowerCase(),e=e.toLowerCase());(s=t.indexOf(e,r))>-1;)a.push({position:s,word:e}),r=s+n;return a}d=d.concat(t(e,h,!1)),g=g.concat(t(e,u,!1))}),(d.length>0||g.length>0)&&(a=!0,i=d.length+g.length)),a){[d,g].forEach(function(e){e.sort(function(e,t){return t.position!==e.position?t.position-e.position:e.word.length-t.word.length})});var v=[];0!=d.length&&v.push(r(l,0,l.length,d));for(var $=[];0!=g.length;){var C=g[g.length-1],m=C.position,x=C.word,w=m-20,y=m+80;0>w&&(w=0),y<m+x.length&&(y=m+x.length),y>p.length&&(y=p.length),$.push(r(p,w,y,g))}$.sort(function(e,t){return e.searchTextCount!==t.searchTextCount?t.searchTextCount-e.searchTextCount:e.hits.length!==t.hits.length?t.hits.length-e.hits.length:e.start-t.start});var T=parseInt("1");T>=0&&($=$.slice(0,T));var b="";b+=0!=v.length?"<li><a href='"+f+"' class='search-result-title'>"+s(l,v[0])+"</a>":"<li><a href='"+f+"' class='search-result-title'>"+l+"</a>",$.forEach(function(e){b+="<a href='"+f+'\'><p class="search-result">'+s(p,e)+"...</p></a>"}),b+="</li>",o.push({item:b,searchTextCount:c,hitCount:i,id:o.length})}}),1===t.length&&""===t[0])s.innerHTML='<div id="no-result"><i class="fa fa-search fa-5x"></i></div>';else if(0===o.length)s.innerHTML='<div id="no-result"><i class="fa fa-frown-o fa-5x"></i></div>';else{o.sort(function(e,t){return e.searchTextCount!==t.searchTextCount?t.searchTextCount-e.searchTextCount:e.hitCount!==t.hitCount?t.hitCount-e.hitCount:t.id-e.id});var a='<ul class="search-result-list">';o.forEach(function(e){a+=e.item}),a+="</ul>",s.innerHTML=a}};r.addEventListener("input",a),$(".local-search-pop-overlay").remove(),$("body").css("overflow",""),proceedsearch()}})};$(".popup-trigger").click(function(e){e.stopPropagation(),isfetched===!1?searchFunc(path,"local-search-input","local-search-result"):proceedsearch()}),$(".popup-btn-close").click(onPopupClose),$(".popup").click(function(e){e.stopPropagation()}),$(document).on("keyup",function(e){var t=27===e.which&&$(".search-popup").is(":visible");t&&onPopupClose()})</script><script>function addCount(Counter) {
      var $visitors = $('.leancloud_visitors');
      var url = $visitors.attr('id').trim();
      var title = $visitors.attr('data-flag-title').trim();

      Counter('get', '/classes/Counter', { where: JSON.stringify({ url }) })
        .done(function({ results }) {
          if (results.length > 0) {
            var counter = results[0];
            
              var $element = $(document.getElementById(url));
              $element.find('.leancloud-visitors-count').text(counter.time + 1);
            
            Counter('put', '/classes/Counter/' + counter.objectId, JSON.stringify({ time: { '__op': 'Increment', 'amount': 1 } }))
            
              .fail(function ({ responseJSON }) {
                console.log(`Failed to save Visitor num, with error message: ${responseJSON.error}`);
              })
          } else {
            
              Counter('post', '/classes/Counter', JSON.stringify({ title: title, url: url, time: 1 }))
                .done(function() {
                  var $element = $(document.getElementById(url));
                  $element.find('.leancloud-visitors-count').text(1);
                })
                .fail(function() {
                  console.log('Failed to create');
                });
            
          }
        })
        .fail(function ({ responseJSON }) {
          console.log(`LeanCloud Counter Error: ${responseJSON.code} ${responseJSON.error}`);
        });
    }
    

    $(function() {
      $.get('https://app-router.leancloud.cn/2/route?appId=' + 'w5RAfQXjT6iFfBtGwiTfxQOP-gzGzoHsz')
        .done(function({ api_server }) {
          var Counter = function(method, url, data) {
            return $.ajax({
              method: method,
              url: 'https://' + api_server + '/1.1' + url,
              headers: {
                'X-LC-Id': 'w5RAfQXjT6iFfBtGwiTfxQOP-gzGzoHsz',
                'X-LC-Key': 'h55viQ2cLIrigHjPHDfIS2ly',
                'Content-Type': 'application/json',
              },
              data: data
            });
          };
          
            addCount(Counter);
          
        });
    });</script><script>!function(){var t=document.createElement("script"),s=window.location.protocol.split(":")[0];"https"===s?t.src="https://zz.bdstatic.com/linksubmit/push.js":t.src="http://push.zhanzhang.baidu.com/push.js";var e=document.getElementsByTagName("script")[0];e.parentNode.insertBefore(t,e)}()</script><script>wpac_init=window.wpac_init||[],wpac_init.push({widget:"Rating",id:16811,el:"wpac-rating",color:"fc6423"}),function(){if(!("WIDGETPACK_LOADED"in window)){WIDGETPACK_LOADED=!0;var e=document.createElement("script");e.type="text/javascript",e.async=!0,e.src="//embed.widgetpack.com/widget.js";var t=document.getElementsByTagName("script")[0];t.parentNode.insertBefore(e,t.nextSibling)}}()</script><style>.copy-btn{display:inline-block;padding:6px 12px;font-size:13px;font-weight:700;line-height:20px;color:#333;white-space:nowrap;vertical-align:middle;cursor:pointer;background-color:#eee;background-image:linear-gradient(#fcfcfc,#eee);border:1px solid #d5d5d5;border-radius:3px;user-select:none;outline:0}.highlight-wrap .copy-btn{transition:opacity .3s ease-in-out;opacity:0;padding:2px 6px;position:absolute;right:4px;top:8px}.highlight-wrap .copy-btn:focus,.highlight-wrap:hover .copy-btn{opacity:1}.highlight-wrap{position:relative}</style><script>$(".highlight").each(function(e,t){var n=$("<div>").addClass("highlight-wrap");$(t).after(n),n.append($("<button>").addClass("copy-btn").append("复制").on("click",function(e){var t=$(this).parent().find(".code").find(".line").map(function(e,t){return $(t).text()}).toArray().join("\n"),n=document.createElement("textarea"),o=document.createRange(),a=window.getSelection(),i=window.pageYOffset||document.documentElement.scrollTop;n.style.top=i+"px",n.style.position="absolute",n.style.opacity="0",n.value=t,n.textContent=t,n.contentEditable=!0,n.readOnly=!1,document.body.appendChild(n),o.selectNode(n),a.removeAllRanges(),a.addRange(o),n.setSelectionRange(0,t.length);var d=document.execCommand("copy");d?$(this).text("复制成功"):$(this).text("复制失败"),n.blur(),$(this).blur()})).on("mouseleave",function(e){var t=$(this).find(".copy-btn");setTimeout(function(){t.text("复制")},300)}).append(t)})</script><script src="/live2d/autoload.js"></script></body></html><!-- rebuild by neat -->